### PR TITLE
[AU4] Added effect preview function as is. Step 1

### PR DESCRIPTION
**Pull request by igorkorsukov**

### CODE REVIEW

File: au4/src/effects/effects_base/qml/Audacity/Effects/EffectsViewerDialog.qml [hunk 1]
```diff
@@ -58,5 +49,27 @@ StyledDialogView {
         Component.onCompleted: {
             bbox.buttonById(ButtonBoxModel.Apply).enabled = false
         }
+
```
Comment (saintmatthieu) [inline au4/src/effects/effects_base/qml/Audacity/Effects/EffectsViewerDialog.qml:52]: There was a request from Product to have the `Apply` button being the default one and with blue highlight, like in the save-on-close dialog. Would you please add this?

File: au4/src/effects/effects_base/internal/effectsprovider.cpp [hunk 2]
```diff
@@ -182,3 +187,241 @@ muse::Ret EffectsProvider::performEffect(AudacityProject& project, Effect* effec
 
     return success ? muse::make_ok() : muse::make_ret(Ret::Code::UnknownError);
 }
+
```
Comment (saintmatthieu) [inline au4/src/effects/effects_base/internal/effectsprovider.cpp:190]: I can assume that this is an unmodified copy-paste (except for formatting), right? Hence I won't really review its content.

**Review by saintmatthieu — APPROVED** [review]

File: au4/src/effects/effects_base/internal/effectsprovider.cpp (hunk 2 above)
Comment (igorkorsukov) [reply au4/src/effects/effects_base/internal/effectsprovider.cpp:190]: Right!

**Comment by saintmatthieu** [comment]
I just notice the `Au3Player::makeTransportTracks` implementation. Please consider re-using.

**Comment by igorkorsukov** [comment]
> I just notice the `Au3Player::makeTransportTracks` implementation. Please consider re-using.

Yes, thank you, I will do this in the second step, during the refactoring stage

**Pull request closed without merging**